idaes/apps/grid_integration/coordinator.py

```diff
@@ -0,0 +1,868 @@
+        ]
+
+        tracking_horizon = len(self.projection_tracker.time_set)
+
+        market_signals = self._assemble_sced_tracking_market_signals(
+            gen_name=gen_name,
+            hour=hour,
+            sced_dispatch=None,
+            tracking_horizon=tracking_horizon,
+            current_ruc_dispatch_dicts=current_ruc_dispatch_dicts,
+            next_ruc_dispatch_dicts=None,
+        )
+        return market_signals
+
+    def project_tracking_trajectory(self, options, simulator, ruc_hour):
+
+        """
+        This function projects the full power dispatch trajectory from the
+        tracking model so we can use it to update the bidding model, i.e. advance
+        the time for the bidding model.
+
+        Arguments:
+            options: Prescient options from prescient.simulator.config.
+
+            simulator: Prescient simulator.
+
+            ruc_hour: the hour RUC is being solved
+
+        Returns:
+            full_projected_trajectory: the full projected power dispatch trajectory.
+        """
+
+        current_date = simulator.time_manager.current_time.date
+        current_hour = simulator.time_manager.current_time.hour
+
+        self._clone_tracking_model()
+
+        for hour in range(ruc_hour, 24):
+
+            # assemble market_signals
+            market_signals = self.assemble_project_tracking_signal(
+                options=options, simulator=simulator, hour=hour
+            )
+            # solve tracking
+            self.projection_tracker.track_market_dispatch(
+                market_dispatch=market_signals, date=current_date, hour=current_hour
+            )
+
+        # merge the trajectory
+        full_projected_trajectory = {}
+        for stat in self.tracker.daily_stats:
+            full_projected_trajectory[stat] = self.tracker.daily_stats.get(
+                stat
+            ) + self.projection_tracker.daily_stats.get(stat)
+
+        # clear the projection stats
+        self.projection_tracker.daily_stats = None
+
+        return full_projected_trajectory
+
+    def _clone_tracking_model(self):
+        """
+        Clone the model in tracker and replace that of projection tracker. In this
+        way, tracker and projection tracker have the same states before projection.
+
+        Arguments:
+            None
+
+        Returns:
+            None
+        """
+
+        # iterate all the variables and params and clone the values
+        objects_list = [pyo.Var, pyo.Param]
+        for obj in objects_list:
+            for tracker_obj, proj_tracker_obj in zip(
+                self.tracker.model.component_objects(obj),
+                self.projection_tracker.model.component_objects(obj),
+            ):
+                for idx in tracker_obj.index_set():
+                    if pyo.value(proj_tracker_obj[idx]) != pyo.value(tracker_obj[idx]):
+                        proj_tracker_obj[idx] = round(pyo.value(tracker_obj[idx]), 4)
```

Comment:
@jsiirola what about something like this instead?

```suggestion
            for proj_tracker_obj in self.projection_tracker.model.component_objects(obj):
                tracker_obj = self.projection_tracker.model.find_component(proj_tracker_obj.name)
                for idx in proj_tracker_obj.index_set():
                    if pyo.value(proj_tracker_obj[idx]) != pyo.value(tracker_obj[idx]):
                        proj_tracker_obj[idx] = round(pyo.value(tracker_obj[idx]), 4)
```

Reply:
That would work, too.  I think I would go with the `find_component()` route if you expect that one model may have more objects than the other.  Otherwise, I would expect that the `sort=` would be more efficient.  We should note that the `find_component` route will be very slow through Pyomo 6.4.  Starting in 6.5 thinkgs will be faster (but I would expect still slower than the `sort=` route).

If we go the `sort=` route, I would also recommend using `zip_longest` and not `zip` (zip will quit when the first list is exhausted, so if the models are different sizes, we wouldn't see an error)

Reply:
> If we go the `sort=` route, I would also recommend using `zip_longest` and not `zip` (zip will quit when the first list is exhausted, so if the models are different sizes, we wouldn't see an error), and we should absolutely compare the component names to make sure that things are lining up correctly.

I like this. I suspect there's no reason to think the models will diverge, but at least this way we will get a reasonable error if it happens.

Reply:
I just pushed to address this! Could you take a look at it @bknueven ? Thanks!

Reply:
This looks good!